crates/bin/c8y_configuration_poc/src/mqtt_ext.rs

```diff
@@ -0,0 +1,180 @@
+use async_trait::async_trait;
+use mqtt_channel::MqttError;
+use mqtt_channel::SinkExt;
+use mqtt_channel::StreamExt;
+use mqtt_channel::TopicFilter;
+use tedge_actors::mpsc::channel;
+use tedge_actors::mpsc::Receiver;
+use tedge_actors::mpsc::Sender;
+use tedge_actors::Actor;
+use tedge_actors::ActorBuilder;
+use tedge_actors::ChannelError;
+use tedge_actors::DynSender;
+use tedge_actors::LinkError;
+use tedge_actors::MessageBox;
+use tedge_actors::PeerLinker;
+use tedge_actors::RuntimeError;
+use tedge_actors::RuntimeHandle;
+
+pub type MqttMessage = mqtt_channel::Message;
+
+#[derive(Debug)]
+pub enum MqttActorMessage {
+    DirectMessage(mqtt_channel::Message),
+    PeerMessage(mqtt_channel::Message),
+}
+
+pub struct MqttActorBuilder {
+    pub mqtt_config: mqtt_channel::Config,
+    pub publish_channel: (Sender<MqttMessage>, Receiver<MqttMessage>),
+    pub subscriber_addresses: Vec<(TopicFilter, DynSender<MqttMessage>)>,
+}
+
+impl MqttActorBuilder {
+    pub fn new(config: mqtt_channel::Config) -> Self {
+        MqttActorBuilder {
+            mqtt_config: config,
+            publish_channel: channel(10),
```

Comment:
The `channel(10)` means it can have 10 messages at any time.

Reply:
Yes. And if the sender try to send more, it will be blocked till there is room for its message.